Test.Activities.Automation.TimerJob/TimerJob.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.SharePoint.Administration;
+using Test.Activities.Automation.ActivityLib.Models;
+using Test.Activities.Automation.ActivityLib.Models.Helpers;
+using Test.Activities.Automation.ActivityLib.Utils.Constants;
+
+namespace Test.Activities.Automation.TimerJob
+{
+    public class TimerJob : SPJobDefinition
+    {
+        private const int RequestAttempts = 3;
+
+        private ILogger _logger;
+
+        public TimerJob()
+        {
+            try
+            {
+                _logger = new ULSLogger(GetType().FullName);
+            }
+            catch
+            {
+            }
+        }
+
+        public TimerJob(string name, SPWebApplication webApp, SPServer server, SPJobLockType lockType)
+            : base(name, webApp, server, lockType)
+        {
+        }
+
+        public override async void Execute(Guid targetInstanceId)
+        {
+            try
+            {
+                _logger?.LogInformation("Timer executing");
+
+                try
+                {
+                    var activityService = new FetchActivityService(_logger);
+
+                    var activities = activityService.FetchActivities();
+
+                    await SendActivities(activities);
+                }
+                catch (Exception e)
+                {
+                    _logger?.LogError(e.Message);
+                }
+
+                _logger?.LogInformation("Timer has executed");
+            }
+            catch
+            {
+            }
+        }
+
+        private async Task SendActivities(IEnumerable<ActivityInfo> activities)
+        {
+            try
+            {
+                _logger?.LogInformation("Sending activities to service");
+
+                var str = APIHelper.JsonSerialize(activities);
+                var uri = new Uri(Constants.ServiceUrl);
+
+                for (var i = 0; i < RequestAttempts; i++)
+                {
+                    if (await APIHelper.PostJsonAsync(uri, str) == HttpStatusCode.OK) return;
```

Comment:
In case of exception it bubbles up to

Reply:
In case of exception it bubbles up from PostJsonAsync() and is handled by SendActivities() in the catch block. The exception handling will not work properly when using async method without "await" keyword or void async method.

Reply:
I mean if status code is not 200. For an instance, if service URL is not valid, etc.